---
title: "Mouse Lingon diet experiment: GO annotation"
author:
   name: "Shuyi Li"
   email: shuyi.li@med.lu.se
   affiliation: LUDC Bioinformatics Unit
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE, warning=FALSE, eval=FALSE)
knitr::opts_knit$set(root.dir = '../')
```

## Libraries

```{r,message=FALSE}
#load packages
library(org.Mm.eg.db)
library(gprofiler2)
library(dplyr)
```

## GO annotation

### Load data

```{r}
pattern_tab <- read.table("Matrix/GeneMat_HFD_Lingon_LDF.Ebseqresults", header = TRUE)
pattern_tab <- tibble::rownames_to_column(pattern_tab, "gene")
mean_tab <- read.table("Matrix/GeneMat_HFD_Lingon_LDF.Ebseqresults.condmeans", header = TRUE)
mean_tab <- tibble::rownames_to_column(mean_tab, "gene")
names(mean_tab) <- c("gene","HDF","Lingon","LFD")
norm_tab <- read.table("Matrix/GeneMat_HFD_Lingon_LDF.Ebseqresults.normalized_data_matrix", header = TRUE)
norm_tab <- tibble::rownames_to_column(norm_tab, "gene")
full_gene <- read.table("Matrix/full_gene_list.csv", header = FALSE)
names(full_gene) <- c("EnsemblID","GeneSymbol")
```

### Merge data

```{r}
full_info_tab <- left_join(mean_tab, norm_tab, by = "gene")
full_info_tab <- full_join(full_info_tab,pattern_tab, by = "gene")
full_info_tab <- cbind(full_info_tab,full_gene)
```

### Convert Ensemblid to GO id

```{r}
id_go <- gconvert(full_gene[,1], organism="mmusculus", target="GO")
write.table(id_go, "Matrix/Ensemble_GO.csv", row.names = F)
write.table(full_info_tab, "Matrix/GeneMat_HFD_Lingon_LDF.Ebseqresults.fullinfo",
            row.names = F)
```

### GO annotation using goatools

Download GO reference:

```{bash}
wget http://geneontology.org/ontology/go-basic.obo
mv go-basic.obo reference/
```

GO annotation:

```{bash}
scr/go_parser.py \
   -i Matrix/GeneMat_HFD_Lingon_LDF.Ebseqresults.fullinfo \
   -a Matrix/Ensemble_GO.csv \
   -o results/GeneMat_HFD_Lingon_LDF.Ebseqresults.GOannotation
```

## Subset 

Subset the differentially expressed genes (FDR=0.05) from this dataset:

```{bash}
awk '$26>=0.95 {print $0}' \
   results/enrichment/GeneMat_HFD_Lingon_LDF.Ebseqresults.GOannotation \
   > results/enrichment/GeneMat_HFD_Lingon_LDF.Ebseqresults.GOannotation_FDR0.05
```

## Interesting functions

Subset the differentially expressed genes (FDR=0.05) that have terms including 
words `glucose`, `adipo`, `mitoch`:

```{bash}
grep -E "glucose|adipo|mitoch" \
   results/enrichment/GeneMat_HFD_Lingon_LDF.Ebseqresults.GOannotation_FDR0.05 \
   > results/enrichment/GeneMat_HFD_Lingon_LDF.Ebseqresults.GOannotation_FDR0.05_glucose-adipo-mitoch
```

Nothing found.
