---
title: "Mouse Lingon diet experiment: Enrichment Analysis"
author:
   name: "Shuyi Li"
   email: shuyi.li@med.lu.se
   affiliation: LUDC Bioinformatics Unit
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    keep_md: true
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE, warning=FALSE, eval=TRUE, fig.width=10)
knitr::opts_knit$set(root.dir = '../')
```

## Libraries

```{r, message=FALSE}
library(clusterProfiler)
library(org.Mm.eg.db)
library(ggplot2)
library(gprofiler2)
```

## Separate genes based on different patterns

Input for `run_select_pattern.sh`: RSEM result (GeneMat_HFD_Lingon_LDF.Ebseqresults/GeneMat_HFD_Lingon_LDF.Ebseqresults_FDR_0.05.tab)

Output: Gene list with EnsemblID	and GeneSymbol, classified as different patterns
(Output folder: Matrix/pattern)

```{bash, eval=FALSE}
#seperate genes based on different patterns
scr/run_select_pattern.sh
#full list of expressed genes
cut -f 1 Matrix/GeneMat_HFD_Lingon_LDF.Ebseqresults | \
   sed 's/_/\t/;s/"//g' | \
   tail -n +2 \
   > Matrix/full_gene_list2.csv
```

## Load data

```{r}
pattern2.FDR0.05 <- read.table("Matrix/pattern/pattern2_FDR0.05.csv",
                               header = TRUE)
pattern3.FDR0.05 <- read.table("Matrix/pattern/pattern3_FDR0.05.csv",
                               header = TRUE)
pattern4.FDR0.05 <- read.table("Matrix/pattern/pattern4_FDR0.05.csv",
                               header = TRUE)
pattern_table <- read.table(gzfile("Matrix/GeneMat_HFD_Lingon_LDF.Ebseqresults.pattern.gz"),
                            header = TRUE)
full_gene <- read.table("Matrix/full_gene_list.csv", header = FALSE)
names(full_gene) <- c("EnsemblID","GeneSymbol")
```

## RSEM result summary

Summarize the number of genes classified to each pattern (FDR cutoff = 0.05)

```{r}
names(pattern_table) <- c("HDF","Lingon","LFD")
RSEM_summary <-data.frame(c(0,155,40,123,0,318),
                 row.names = c("pattern1.FDR0.05","pattern2.FDR0.05",
                               "pattern3.FDR0.05","pattern4.FDR0.05",
                               "pattern5.FDR0.05","Sum"))
names(RSEM_summary) <- c("number of genes")
knitr::kable(pattern_table)
knitr::kable(RSEM_summary)
```

Explanation:

- **Pattern1** - no differential change between any of the experiments.
- **Pattern2** - differentially expressed (DE) in LFD compared to HFD & Lingon.
- **Pattern3** - DE in Lingon compared to LFD & HFD.
- **Pattern4** - DE in HFD compared to Lingon & LFD.

## ClusterProfiler

[ClusterProfiler](https://guangchuangyu.github.io/software/clusterProfiler)

```{r}
ego <- function(pattern_data){
   ego_pattern <- enrichGO(gene = pattern_data[,1], # test gene list
                           universe = full_gene[,1], # background gene list
                           OrgDb = org.Mm.eg.db, 
                           keyType = "ENSEMBL", 
                           ont = 'ALL',       
                           pAdjustMethod = "BH",
                           pvalueCutoff  = 0.01,
                           qvalueCutoff  = 0.05,
                           readable      = TRUE)
   file_name <- paste("results/clusterprofiler/",
                      deparse(substitute(pattern_data)),
                      "_GO_enrich.csv", sep = "")
   write.table(as.data.frame(ego_pattern), file_name, row.names = F, sep = "\t")
   return(ego_pattern)
}

full_entrez <- bitr(full_gene[,1],
                    fromType = "ENSEMBL",
                    toType = "ENTREZID",
                   OrgDb = org.Mm.eg.db)

eKEGG <- function(pattern_data){
   pattern_entrez <- bitr(pattern_data[,1],
                    fromType = "ENSEMBL",
                    toType = c("ENTREZID"),
                    OrgDb = org.Mm.eg.db)
   eKEGG_pattern <- enrichKEGG(gene = pattern_entrez[,2], # test gene list
                           universe = full_entrez[,2], # background gene list
                           organism='mmu',
                           pvalueCutoff  = 0.01,
                           qvalueCutoff  = 0.05)
   eKEGG_pattern_genes <- setReadable(eKEGG_pattern,
                                      OrgDb = org.Mm.eg.db,
                                      keyType="ENTREZID")
   file_name <- paste("results/clusterprofiler/",
                      deparse(substitute(pattern_data)),
                      "_KEGG_enrich.csv", sep = "")
   write.table(as.data.frame(eKEGG_pattern_genes),
             file_name, row.names = F, sep = "\t")
   return(eKEGG_pattern_genes)
}
```

### GO pattern2.FDR0.05 (LFD specific)

```{r}
ego_pattern2.FDR0.05 <- ego(pattern2.FDR0.05)
ego_pattern2.FDR0.05_n  <- dim(ego_pattern2.FDR0.05)[1]
ego_pattern2.FDR0.05_n
```

Gene counts and significance of enrichment:

```{r}
barplot(ego_pattern2.FDR0.05, showCategory=ego_pattern2.FDR0.05_n)
```

Gene ratio, counts and significance of enrichment:

```{r}
dotplot(ego_pattern2.FDR0.05, showCategory=ego_pattern2.FDR0.05_n)
```

Differentially expressed genes that belong enriched terms:

```{r}
heatplot(ego_pattern2.FDR0.05)
```

### GO pattern3.FDR0.05 (Lingon specific)

```{r}
ego_pattern3.FDR0.05 <- ego(pattern3.FDR0.05)
ego_pattern3.FDR0.05_n  <- dim(ego_pattern3.FDR0.05)[1]
ego_pattern3.FDR0.05_n
```

Gene counts and significance of enrichment:

```{r}
barplot(ego_pattern3.FDR0.05,showCategory=ego_pattern3.FDR0.05_n,drop=T)
```

Gene ratio, counts and significance of enrichment:

```{r}
dotplot(ego_pattern3.FDR0.05,showCategory=ego_pattern3.FDR0.05_n)
```

Differentially expressed genes that belong enriched terms:

```{r}
heatplot(ego_pattern3.FDR0.05)
```

### GO pattern4.FDR0.05 (HFD specific)

```{r}
ego_pattern4.FDR0.05 <- ego(pattern4.FDR0.05)
ego_pattern4.FDR0.05_n  <- dim(ego_pattern4.FDR0.05)[1]
ego_pattern4.FDR0.05_n
```

Gene counts and significance of enrichment:

```{r}
barplot(ego_pattern4.FDR0.05,showCategory=ego_pattern4.FDR0.05_n,drop=T)
```

Gene ratio, counts and significance of enrichment:

```{r}
dotplot(ego_pattern4.FDR0.05,showCategory=ego_pattern4.FDR0.05_n)
```

Differentially expressed genes that belong enriched terms:

```{r}
heatplot(ego_pattern4.FDR0.05)
```


### KEGG pattern2.FDR0.05 (LFD specific)

```{r, message=F}
eKEGG_pattern2.FDR0.05 <- eKEGG(pattern2.FDR0.05)
knitr::kable(as.data.frame(eKEGG_pattern2.FDR0.05))
```

### KEGG pattern3.FDR0.05 (Lingon specific)

```{r, message=F}
eKEGG_pattern3.FDR0.05 <- eKEGG(pattern3.FDR0.05)
knitr::kable(as.data.frame(eKEGG_pattern3.FDR0.05))
```

### KEGG pattern4.FDR0.05 (HFD specific)

```{r, message=F}
eKEGG_pattern4.FDR0.05 <- eKEGG(pattern4.FDR0.05)
knitr::kable(as.data.frame(eKEGG_pattern4.FDR0.05))
```

## gProfiler

[gProfiler2](https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html)

```{r}
namesP <- c("term_id", "source", "p_value", 'term_size', 'intersection_size',
            "term_name","intersection")
ggo <- function(pattern_data){
   gost_pattern <- gost(pattern_data[,2],
                        domain_scope = "custom",
                        custom_bg = full_gene[,2],
                        sources = c('GO', 'KEGG', 'REAC', 'WP'),
                        organism ='mmusculus',
                        significant = TRUE,
                        evcodes=TRUE)
   file_name <- paste("results/gprofiler/",
                      deparse(substitute(pattern_data)),
                      "_gProfiler_enrich.csv", sep = "")
   write.table(as.data.frame(gost_pattern$result[,namesP]),
            file_name, row.names = F, sep = "\t")
   return(gost_pattern)
}
```

### pattern2.FDR0.05 (LFD specific)

```{r}
gost_pattern2 <- ggo(pattern2.FDR0.05)
gostplot(gost_pattern2)
```

### pattern3.FDR0.05 (Lingon specific)

```{r}
gost_pattern3 <- ggo(pattern3.FDR0.05)
gostplot(gost_pattern3)
```

### pattern4.FDR0.05 (HFD specific)

```{r}
gost_pattern4 <- ggo(pattern4.FDR0.05)
gostplot(gost_pattern4)
```

## Results tables

- `results/clusterprofiler/` - clusterProfiler GO and KEGG enrichment results.
- `results/gprofiler/` - gProfiler GO, KEGG, Reactome, and WikiPathways enrichment results.
